hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,128 @@
+defmodule Hangman.Game do
+
+  defmodule State do
+
+    defstruct(
+      game_state:    :initializing,
+      turns_left:    7,
+      answer:        "",
+      letters:       [],
+      used:          [],
+      last_guess:    ""
+    )
+
+  end
+
+  def new_game() do
+    word = Dictionary.random_word()
+    %State{answer: word, letters: fill_in_word(word, [])}
+  end
+
+  def tally(game) do
+    %{
+      game_state:     game.game_state,
+      turns_left:     game.turns_left,
+      letters:        game.letters,
+      used:           game.used,
+      last_guess:     game.last_guess
+    }
+  end
+
+  def make_move(game, guess) do
+
+    handle_guess(game, guess, game.last_guess,
+      first_matching_index(game.answer, guess),
+      first_blank_index(fill_in_word(game.answer, game.used ++ [ guess ])),
+      game.turns_left)
+
+  end
+
+ # Player loses
+  defp handle_guess(game, guess, _, nil, _, 1) do
```

Comment:
## -5 duplication

Have a look at the `handle_guess` functions below. Can you see duplication? How could you minimize it?